File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
     List<List<Integer>> pathValues = new ArrayList<>();
    public int sumNumbers(TreeNode root) {
        if (root == null) return 0;
        List<Integer> path = new ArrayList<>();
        dfs(root, path);
       
        int sum = 0;
        for (List<Integer> pathValue : pathValues) {
            int pathSum = 0;
            for (int i = 0; i < pathValue.size(); i++) {
                pathSum += pathValue.get(i) * Math.pow(10, pathValue.size() - i - 1);
            }
            sum += pathSum;
        }
        return sum;

    }

    private void dfs(TreeNode root, List<Integer> path) {
        if (root == null) return;
        path.add(root.val);
        if (root.left == null && root.right == null) {
            pathValues.add(new ArrayList<>(path));
        }
        dfs(root.left, path);
        dfs(root.right, path);
        path.remove(path.size() - 1);
    }
}

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        self.arr=[]
        def dfs(node,ls):
            if node:
                if not node.left and not node.right:
                    self.arr.append(ls+[node.val])
                dfs(node.left,ls+[node.val])
                dfs(node.right,ls+[node.val])
        dfs(root,[])
   
        def convert(list): 
            s = [str(i) for i in list] 
            res = int("".join(s))
            return(res)
        ans=0
        for x in self.arr:
            ans+=convert(x)
        return ans 